app/_src/gateway/plugin-development/wasm.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Create a Proxy-Wasm filter
+book: plugin_dev
+chapter: 11
+---
+
+## Your first Proxy-Wasm filter
+
+Filters can be written as a compilation target in almost any language that supports WebAssembly, but the easiest way to get started is by
+using a language which has an existing [Proxy-Wasm](https://github.com/proxy-wasm/spec)
+SDK, such as [Go](https://github.com/tetratelabs/proxy-wasm-go-sdk/)
+or [Rust](https://github.com/proxy-wasm/proxy-wasm-rust-sdk/).
+
+## Coding, building and deploying a Proxy-Wasm filter
+
+Kong has made Go and Rust Proxy-Wasm filter templates available. They can be used
+to start writing your own filters, or to build as they are and get your first `hello
+world`-type filter:
+
+* [Go Proxy-Wasm filter template](https://github.com/Kong/proxy-wasm-go-filter-template/)
+* [Rust Proxy-Wasm filter template](https://github.com/Kong/proxy-wasm-rust-filter-template/)
+
+### Deploying the filter with Kong
+
+After writing and building the filter, the next steps are to configure Kong to enable Wasm support and add the filter, then associate the filter to a route or a service. You will then be able to start Kong and issue requests that go through the filter.
+
+#### Configuration
+
+WebAssembly support must be enabled in {{site.base_gateway}}:
+
+```console
+$ export KONG_WASM=on
+```
+
+Additionally, the `wasm_filters_path` parameter must be configured in order for
+{{site.base_gateway}} to load the filter at runtime. This can be any folder
+containing `.wasm` files.
+
+During local development,
+when a short feedback loop is desired, you can set this parameter to
+your build toolchain's output directory (wherever the compiled
+`<filter-name>.wasm` file is produced):
+
+```console
+$ export KONG_WASM_FILTERS_PATH=/path/to/my_filter/build
+```
+
+#### Link to a Kong service and route
+
+Now, the next step is to create a Kong filter chain entity using the
+filter created in the previous step and associate it with a Kong route
+or service. This will be done by creating a declarative yaml config file
+at `kong.yml`:
+
+```yaml
+---
+_format_version: '3.0'
+services:
+  - name: my-wasm-service
+    url: http://mockbin.org
+    routes:
+      - name: my-wasm-route
+        paths:
+          - /
+    filter_chains:
+      - name: my-wasm-demo
+        filters:
+          - name: my_filter
+            config: >-
+              {
+                "my_greeting": "Hello from Kong using WebAssembly!"
+              }
+```
+
+```console
+$ export KONG_DATABASE=off
```

Comment:
Do we have any document for using wasmx filter when `KONG_DATABASE=on`? I did a try to set this up with database enabled, but failed.

Reply:
Just a note: we don't call Lua Plugins "OpenResty plugins", the same way these are "Wasm filters", not "WasmX filters".

Reply:
> Do we have any document for using wasmx filter when KONG_DATABASE=on?

We don't, but I have used it with a database (and the `spec/` tests do exercise it with a DB as well). @zhongweiy maybe your migrations were out of sync?

Reply:
@hishamhm, because I don't know how to translate each step into Admin API call, i.e. what is the Admin API command to do the step "Link to a Kong service and route"?